nx/lib/nx/tensor.ex

```diff
@@ -210,48 +210,68 @@ defmodule Nx.Tensor do
     raise "Access.pop/2 is not yet supported by Nx.Tensor"
   end
 
-  defimpl Inspect do
+  defimpl Inspect, for: Nx.Tensor do
     import Inspect.Algebra
 
-    def inspect(
-          %{shape: shape, names: names, type: type, vectorized_axes: vectorized_axes} = tensor,
-          opts
-        ) do
+    defp format_stats(tensor) do
```

Comment:
Why are these specific stats chosen? Also, why not calculate them over each axis?

I think these are pretty meaningless for certain types of tensors. Identity, permutation, orthogonal tensors, for instance.